File: backtracking/78_Subsets.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
 var subsets = function(nums) {
    let length = nums.length;
    let result = [];

    for(let i=0; i < 2 ** length; i++){    
        let sub = []
        for(let j =0 ; j < length; j++){
            if((i & (1 << j) ) > 0){               
                sub.push(nums[j])
            }
        }
        result.push(sub)
    }     
    
    return result
};

File: backtracking/78_Subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        length = len(nums)
        size = pow(2, length)
        result = []
        for i in range(size):
            sub = []

            for j in range(length):
                if i & (1 << j) > 0:
                    sub.append(nums[j])
                    
            result.append(sub)

        return result
